examples/shader/custom_render_phase.rs

```diff
@@ -0,0 +1,630 @@
+        layout: &MeshVertexBufferLayoutRef,
+    ) -> Result<RenderPipelineDescriptor, SpecializedMeshPipelineError> {
+        // We will only use the position of the mesh in our shader so we only need to specify that
+        let mut vertex_attributes = Vec::new();
+        if layout.0.contains(Mesh::ATTRIBUTE_POSITION) {
+            // Make sure this matches the shader location
+            vertex_attributes.push(Mesh::ATTRIBUTE_POSITION.at_shader_location(0));
+        }
+        // This will automatically generate the correct `VertexBufferLayout` based on the vertex attributes
+        let vertex_buffer_layout = layout.0.get_layout(&vertex_attributes)?;
+
+        Ok(RenderPipelineDescriptor {
+            label: Some("Specialized Mesh Pipeline".into()),
+            // We want to reuse the data from bevy so we use the same bind groups as the default
+            // mesh pipeline
+            layout: vec![
+                // Bind group 0 is the view uniform
+                self.mesh_pipeline
+                    .get_view_layout(MeshPipelineViewLayoutKey::from(key))
+                    .clone(),
+                // Bind group 1 is the mesh uniform
+                self.mesh_pipeline.mesh_layouts.model_only.clone(),
+            ],
+            push_constant_ranges: vec![],
+            vertex: VertexState {
+                shader: self.shader_handle.clone(),
+                shader_defs: vec![],
+                entry_point: "vertex".into(),
+                buffers: vec![vertex_buffer_layout],
+            },
+            fragment: Some(FragmentState {
+                shader: self.shader_handle.clone(),
+                shader_defs: vec![],
+                entry_point: "fragment".into(),
+                targets: vec![Some(ColorTargetState {
+                    format: TextureFormat::bevy_default(),
+                    blend: None,
+                    write_mask: ColorWrites::ALL,
+                })],
+            }),
+            primitive: PrimitiveState {
+                topology: key.primitive_topology(),
+                front_face: FrontFace::Ccw,
+                cull_mode: Some(Face::Back),
+                polygon_mode: PolygonMode::Fill,
+                ..default()
+            },
+            depth_stencil: None,
+            // It's generally recommended to specialize your pipeline for MSAA,
+            // but it's not always possible
+            multisample: MultisampleState::default(),
+            zero_initialize_workgroup_memory: false,
+        })
+    }
+}
+
+// We will reuse render commands already defined by bevy to draw a 3d mesh
+type DrawMesh3dStencil = (
+    SetItemPipeline,
+    // This will set the view bindings in group 0
+    SetMeshViewBindGroup<0>,
+    // This will set the mesh bindings in group 1
+    SetMeshBindGroup<1>,
+    // This will draw the mesh
+    DrawMesh,
+);
+
+// This is the data required when you define a custom phase in bevy. More specifically this is the
```

Comment:
I think it might be more clear to also mention that this is the data needed to render a single phase item (i.e. an entity)